### PR TITLE
Cleanup comments in pkg/build.

### PR DESCRIPTION
**Pull request by matanlurey**
Some little cleanups I've been stashing while working on `build_test`.

No breaking changes other than `/*<T>*/` to `<T>`. Feel free to nit-pick.

### CODE REVIEW

File: build/lib/src/asset/reader.dart [hunk 1]
```diff
@@ -8,16 +8,27 @@ import 'package:glob/glob.dart';
 
 import 'id.dart';
 
-/// Abstract interface for reading assets.
+/// Standard interface for reading an asset within in a package.
+///
+/// An [AssetReader] is required when implementing the `runBuilder` command.
```
Comment (natebosch) [inline build/lib/src/asset/reader.dart]: when using the `runBuilder` method

File: build/lib/src/asset/reader.dart [hunk 2]
```diff
@@ -8,16 +8,27 @@ import 'package:glob/glob.dart';
...
 abstract class AssetReader {
+  /// Returns a [Future] that completes with the bytes of a binary asset.
+  ///
+  /// * Throws a [PackageNotFoundException] if [id.package] is not found.
```
Comment (natebosch) [inline build/lib/src/asset/reader.dart]: we should double check the bazel implementation to ensure it behaves the same

File: build/lib/src/asset/reader.dart [hunk 3]
```diff
@@ -8,16 +8,27 @@ import 'package:glob/glob.dart';
...
+  /// * Throws a [AssetNotFoundException] if [id.path] is not found.
   Future<List<int>> readAsBytes(AssetId id);
 
+  /// Returns a [Future] that completes the contents of a text asset.
```
Comment (natebosch) [inline build/lib/src/asset/reader.dart]: completes with

File: build/lib/src/asset/writer.dart [hunk 4]
```diff
@@ -6,19 +6,33 @@ import 'dart:convert';
...
 
-/// An [AssetWriter] which tracks all [AssetId]s written during it's lifetime.
+/// An [AssetWriter] which tracks all [assetsWritten] during its lifetime.
 class AssetWriterSpy implements AssetWriter {
   final AssetWriter _delegate;
   final _assetsWritten = new Set<AssetId>();
 
   AssetWriterSpy(this._delegate);
 
+  /// All assets that were written.
```
Comment (natebosch) [inline build/lib/src/asset/writer.dart]: This one isn't adding information

File: build/lib/src/asset/reader.dart (hunk 1 above)
Comment (matanlurey) [inline build/lib/src/asset/reader.dart]: Done.

File: build/lib/src/asset/reader.dart [hunk 5]
```diff
@@ -10,15 +10,15 @@ import 'id.dart';
 
 /// Standard interface for reading an asset within in a package.
 ///
-/// An [AssetReader] is required when implementing the `runBuilder` command.
+/// An [AssetReader] is required when implementing the `runBuilder` method.
```
Comment (natebosch) [inline build/lib/src/asset/reader.dart]: change 'implementing' to 'calling'.

File: build/lib/src/asset/reader.dart (hunk 2 above)
Comment (matanlurey) [inline build/lib/src/asset/reader.dart]: Acknowledged. I'm going to try and add an e2e test in a future PR and I'll try it out.

File: build/lib/src/asset/reader.dart (hunk 3 above)
Comment (matanlurey) [inline build/lib/src/asset/reader.dart]: Done.

File: build/lib/src/asset/writer.dart (hunk 4 above)
Comment (matanlurey) [inline build/lib/src/asset/writer.dart]: Reverted.

File: build/lib/src/asset/reader.dart (hunk 5 above)
Comment (matanlurey) [inline build/lib/src/asset/reader.dart]: Done.

**Pull request merged**